python/GafferUI/PathListingWidget.py

```diff
@@ -585,6 +585,12 @@ def __buttonPress( self, widget, event ) :
 		if event.buttons != event.Buttons.Left :
 			return False
 
+		# A small corner area below the vertical scroll bar may pass through
+		# to us, causing odd selection behavior. Check that we're within the
+		# scroll area.
+		if event.line.p0.x > self._qtWidget().viewport().size().width() or event.line.p0.y > self._qtWidget().viewport().size().height() :
+			return False
```

Comment:
I think this check should be moved to `__indexAt()`, so that it applies to all code attempting to find the index under the mouse. The code immediately below isn't using `__indexAt()`, but I think it should be?

There's also a separate bug in the LightEditor, where a double-click on the corner will give this error :

```
ERROR : EventSignalCombiner : Traceback (most recent call last):
  File "/home/john/dev/build/gaffer-1.0-Python3/python/Gaffer/WeakMethod.py", line 65, in __call__
    return self.__method( s, *args, **kwArgs )
  File "/home/john/dev/build/gaffer-1.0-Python3/python/GafferSceneUI/LightEditor.py", line 289, in __buttonDoubleClick
    self.__editSelectedCells( pathListing )
  File "/home/john/dev/build/gaffer-1.0-Python3/python/GafferSceneUI/LightEditor.py", line 329, in __editSelectedCells
    self.__popup = GafferUI.PlugPopup( edits, warning = warnings )
  File "/home/john/dev/build/gaffer-1.0-Python3/python/GafferUI/PlugPopup.py", line 50, in __init__
    assert( len( plugs ) )
AssertionError
```

You can trigger the same bug by hitting <kbd>Return</kbd> while not having any cells selected, so I think an early out for an empty selection in `__editSelectedCells` will do the trick for both. Could you include that as a separate commit, complete with an entry in Changes.md please?

Reply:
> The code immediately below isn't using __indexAt(), but I think it should be?
It could, but I'm not using it to avoid a second `self._qtWidget().viewport().mapFrom()` call. The expansion arrow check needs `qPoint` so we can't use `__indexAt` by itself, unless we modify what it returns.

Maybe I'm being too picky about performance?

Reply:
Cool, that's much cleaner now. Fixed in 37de55c4517ec3a0cf8cc02a35cb9fb38438132b